user_guide_src/source/installation/upgrade_controllers.rst

```diff
@@ -0,0 +1,65 @@
+Upgrade Controllers
+###############################################################################
+
+.. contents::
+    :local:
+    :depth: 1
+
+
+Documentations
+============================================================
+
+- `Controller Documentation Codeigniter 3.X <http://codeigniter.com/userguide3/general/controllers.html>`_
+- `Controller Documentation Codeigniter 4.X <http://codeigniter.com/user_guide/incoming/controllers.html>`_
+
+
+What has been changed
+============================================================
+- Since Namespaces has been added in CodeIgniter 4, the controllers must be changed to support namespaces
+- Controllers don’t use a constructor any more (to invoke CI “magic”) unless that is part of a base controller you make
+- CI provides Request and Response objects for you to work with - more powerful than the CI3-way
+- If you want a base controller (MY_Controller in CI3), make it where you like, e.g. BaseController extends Controller, and then have your controllers extend it
```

Comment:
```suggestion
- Since namespaces have been added to CodeIgniter 4, the controllers must be changed to support namespaces.
- Controllers don’t use constructors any more (to invoke CI “magic”) unless those are part of base controllers you make.
- CI provides Request and Response objects for you to work with - more powerful than the CI3-way.
- If you want a base controller (``MY_Controller`` in CI3), make it where you like, e.g., BaseController extends Controller, and then have your controllers extend it
```